docs/common-tasks/ssh-keys-for-private-repos.md

```diff
@@ -3,9 +3,83 @@
 ## Forward/Import your SSH Key into the build container to clone private repositories
 
 Certain containers like Jenkins and build may need your private key. This is supported by importing your 
-private key into the container via a volume mount.  
+private key into the container via a volume mount.
 
 To get your private key into the build container, volume mount your key into the container at `/root/.ssh/outrigger.key` 
 and it will be processed accordingly.
 
 `~/.ssh/id_rsa:/root/.ssh/outrigger.key`
+
+## Passphrase Protected Keys
+
+For interactive operations, such as use of the build container, passphrase protected
+keys do not pose a challenge as you can enter the passphrase when prompted. Services
+like Jenkins, however, can prove problematic. A solution to this is to use a container
+running an ssh agent with your key added and unlocked. Your container will be able
+to connect with the ssh agent via the use of a shared volume and the SSH_AUTH_SOCK
+variable that points to it.
+
+### SSH Agent Container
+
+The following is an ssh.yml file that can help set up the necessary containers and
+add your desired keys to it.
+
+```
+# ssh.yml
+# Provides a service for the ssh-agent and a service for adding keys to it
+#
+# Agent will need to be restarted and have key(s) re-added to it any time your
+# docker host is restarted.
+#
+version: '2.1'
```

Comment:
While we are using docker-compose 2.1 some, since it's deprecated do we want to document in it?